Boltdir/site-modules/gplt/tasks/create_control_repo_from_production_env.sh

```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+
+ENVIRONMENT=production
+PUPPET_PROD_ENV_DIR=/etc/puppetlabs/code/environments/$ENVIRONMENT
+BASEDIR=/opt/puppet
+CONTROL_REPO_URL=$BASEDIR/control-repo.git
+CHECKOUT_DIR=/root/control-repo
+
+mkdir -p $BASEDIR
+
+if [ -d $CONTROL_REPO_URL ]; then
+    echo "ERROR: The control repo destination already exists" >&2
+    exit 1
+fi
+
+git init --bare $CONTROL_REPO_URL
```

Comment:
Do we need to do some kind of check to ensure that git is installed?  I know it is on our gatling setups, but not sure if it is an a normal install.

Reply:
I will double check, but since code manager uses git I don't believe a modern PE master can function without git installed already.

Reply:
Even if git is not installed, bolt will return the STDERR result to the user.
```
$ bolt task run   --no-host-key-check   --private-key ~/.ssh/id_rsa-acceptance   --nodes ip-10-227-2-25.amz-dev.puppet.net gplt::create_control_repo_from_production_env
Started on ip-10-227-2-25.amz-dev.puppet.net...
Failed on ip-10-227-2-25.amz-dev.puppet.net:
  The task failed with exit code 1:
  /tmp/6d70bdb9-09ad-4eda-a8c6-376ffa3fc95b/create_control_repo_from_production_env.sh: line 16: git: command not found
  /tmp/6d70bdb9-09ad-4eda-a8c6-376ffa3fc95b/create_control_repo_from_production_env.sh: line 17: git: command not found
  cp: cannot stat ‘/etc/puppetlabs/code/environments/production/*’: No such file or directory
  /tmp/6d70bdb9-09ad-4eda-a8c6-376ffa3fc95b/create_control_repo_from_production_env.sh: line 20: cd: /root/control-repo: No such file or directory
  ERROR: Failed to cd into /root/control-repo

  {
  }
Failed on 1 node: ip-10-227-2-25.amz-dev.puppet.net
Ran on 1 node in 1.80 seconds
```

Reply:
Oops, looks like I forgot to include `set -e`. :man_facepalming: 
Here, this is better now:
```
$ bolt task run   --no-host-key-check   --private-key ~/.ssh/id_rsa-acceptance   --nodes ip-10-227-2-25.amz-dev.puppet.net gplt::create_control_repo_from_production_env
Started on ip-10-227-2-25.amz-dev.puppet.net...
Failed on ip-10-227-2-25.amz-dev.puppet.net:
  The task failed with exit code 127:
  /tmp/1f7dc3b5-c5bd-4fbf-af73-61b3f3882f99/create_control_repo_from_production_env.sh: line 18: git: command not found

  {
  }
Failed on 1 node: ip-10-227-2-25.amz-dev.puppet.net
Ran on 1 node in 1.56 seconds
```